### ISSUE TITLE
Incorrect IDE0044 when NRTs are enabled, the class is generic, and the field is only set by a non-this reference

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn#38921]
Visual Studio 16.3.1 and 16.4p1

IDE0044 should not be reported.

```cs
class GenericClass<T>
{
    // IDE0044 Make field readonly
    private int field;

    public static void Method(GenericClass<T> instance)
    {
        instance.field++;
    }
}
```

Similarly:

```cs
class GenericClass<T>
{
    // IDE0044 Make field readonly
    private int field;

    public struct Helper
    {
        private readonly GenericClass<T> instance;

        public Helper(GenericClass<T> instance)
        {
            this.instance = instance;
        }

        public void Method()
        {
            instance.field++;
        }
    }
}
```

Csproj:
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <LangVersion>latest</LangVersion>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```

**Comment by CyrusNajmabadi** [comment]
Doesn't repro for me on 16.4p1:

![image](https://user-images.githubusercontent.com/4564579/65798764-58205980-e127-11e9-8bf6-8f60c1ece976.png)

**Comment by jnm2** [comment]
I'm reproing on 16.4p1. Did you press Ctrl+. on the line? The default visibility wouldn't show any visual.

To help me quickly see when it was present when I was trimming down the repro, I used this editorconfig:

```ini
root = true

[*]
dotnet_style_readonly_field = true:error
```

**Comment by CyrusNajmabadi** [comment]
Crazy!  And yes, i've got this feature turned on.  I'm also on:

![image](https://user-images.githubusercontent.com/4564579/65799837-ee557f00-e129-11e9-977f-6b96dc62a133.png)

**Comment by jnm2** [comment]
So sorry, I forgot that I had added `<Nullable>enable</Nullable> to the project file. Updated at the top.

**Comment by CyrusNajmabadi** [comment]
Yup.  I repro now.  Fascinating!

**Comment by CyrusNajmabadi** [comment]
Tagging @mavasani .  Note the necessity to update your project file.  This didn't repro for me in a 4.7.2 project...

**Comment by jnm2** [comment]
It repros for me on net472 *where I initially discovered it) as well as netcoreapp3.0.
E.g.
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFramework>net472</TargetFramework>
    <LangVersion>latest</LangVersion>
    <Nullable>enable</Nullable>
  </PropertyGroup>

</Project>
```

**Comment by mavasani** [comment]
Tagging @333fred - seems to be another analyzer affected by https://github.com/dotnet/roslyn/issues/38195

**Comment by mavasani** [comment]
Also tagging @chsienki who is currently investigating #38195

**Comment by CyrusNajmabadi** [comment]
Note: if code has to be updated anyways to get the right nullable-behavior, then i'm far more inclined to say that our choices to try to expose nullable through these extra properties seems less useful.  After all, if you still need to update to support nullable properly then we could have exposed nullable not through properties at all. In both cases people need to update to have their code work in the presence of nullable :)

**Comment by jnm2** [comment]
This is marked 'help wanted' but it looks like it's linked to https://github.com/dotnet/roslyn/issues/38195. Is there something a community member could do to move this along?

**Comment by mavasani** [comment]
Fixing #38195 should fix this issue. @chsienki @333fred do we know which release that fix is targeted for? If that is a some way out, then we can possibly put some workaround in the analyzer.

**Comment by chsienki** [comment]
It was originally planned for 16.4, but it got put on hold because of all the ISymbol/Symbol separation work going on in the compiler. Should be a fairly easy fix once that work is finished though.

**Comment by jnm2** [comment]
Still says up-for-grabs. Does that mean that the larger work is too far away?

**Comment by jasonmalinowski** [comment]
That larger work is now landed. @chsienki what's the next step here then?

**Comment by chsienki** [comment]
Was actually planning on starting #38195 on Monday :) aiming for 16.5 but we'll see on timelines if it makes it in.

**Comment by chsienki** [comment]
Fixed via #41123
